driver-sync/src/test/functional/com/mongodb/client/Socks5ProseTest.java

```diff
@@ -0,0 +1,234 @@
+        assumeTrue(SOCKS_AUTH_ENABLED);
+        mongoClient = MongoClients.create(connectionString);
+        runHelloCommand(mongoClient);
+    }
+
+    @ParameterizedTest(name = "Should connect with valid authentication in proxy settings. ConnectionString: {0}")
+    @MethodSource("validAuthSettings")
+    public void shouldConnectWithValidAuthInProxySettings(final ConnectionString connectionString) {
+        assumeTrue(SOCKS_AUTH_ENABLED);
+        mongoClient = MongoClients.create(buildMongoClientSettings(connectionString));
+        runHelloCommand(mongoClient);
+    }
+
+    private static void assertSocksAuthenticationIssue(final ClusterListener clusterListener,
+                                                       final ArgumentCaptor<ClusterDescriptionChangedEvent> captor) {
+        Mockito.verify(clusterListener, atLeast(1)).clusterDescriptionChanged(captor.capture());
+        List<Throwable> errors = captor.getAllValues().stream()
+                .map(ClusterDescriptionChangedEvent::getNewDescription)
+                .map(ClusterDescription::getServerDescriptions)
+                .flatMap(List::stream)
+                .map(ServerDescription::getException)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        assumeFalse(errors.isEmpty());
+        errors.forEach(throwable -> Assertions.assertEquals(MongoSocketOpenException.class, throwable.getClass()));
+    }
+
+    private static void runHelloCommand(final MongoClient mongoClient) {
+        mongoClient.getDatabase("test").runCommand(new Document("hello", 1));
+    }
+
+    private static ConnectionString buildConnectionString(final String uriPrefix,
+                                                          final boolean directConnection,
+                                                          @Nullable final String proxyUsername,
+                                                          @Nullable final String proxyPassword) {
+        StringJoiner joiner;
+        if (uriPrefix.contains("/?")) {
+            joiner = new StringJoiner("&", "&", "");
+        } else {
+            joiner = new StringJoiner("&", "/?", "");
+        }
+
+        joiner.add("proxyHost=" + PROXY_HOST)
+                .add("proxyPort=" + PROXY_PORT);
+        if (proxyPassword != null && proxyUsername != null) {
+            joiner.add("proxyPassword=" + proxyPassword)
+                    .add("proxyUsername=" + proxyUsername);
+        }
+        if (directConnection) {
+            joiner.add("directConnection=" + true);
+        }
+        return new ConnectionString(uriPrefix + joiner);
+    }
+
+    private static ConnectionString buildConnectionString(final String uriPrefix, final boolean directConnection) {
+        return buildConnectionString(uriPrefix, directConnection, null, null);
+    }
+
+    private static MongoClientSettings buildMongoClientSettings(final ConnectionString connectionString) {
```

Comment:
I may be wrong, but it seems to me that all that happens in this method is:

- we create a new `MongoClientSettings.builder`;
- we apply `connectionString` to it;
- we build `MongoClientSettings`.

Can't the same be achieved with `MongoClientSettings.builder().applyConnectionString(connectionString)`?

Reply:
Absolutely, I agree with you that it can be achieved by using  `applyConnectionString`. It would be simpler and cleaner. 

However, the specification requires us to conduct variations of the tests using equivalent proxy options explicitly. 

This approach ensures compliance with the guidelines that state:
> Drivers MUST run variants of these tests in which the proxy options are substituted for MongoClient options.